ankerctl.py

```diff
@@ -260,84 +371,68 @@ def config_import(env, fd):
-        else:
-            exit("This platform does not support autodetection. Please specify file location")
+        try:
+            if useros == 'Darwin':
+                fd = open(darfileloc, 'r')
+            elif useros == 'Windows':
+                fd = open(winfileloc, 'r')
+            else:
+                log.critical("This platform does not support autodetection. Please specify file location")
+        except FileNotFoundError:
+            log.critical("Failed to import file - check if you are logged into Ankerslicer")
 
     log.info("Loading cache..")
 
     # extract auth token
     cache = libflagship.logincache.load(fd.read())["data"]
-    auth_token=cache["auth_token"]
+    auth_token = cache["auth_token"]
 
     # extract account region
-    region=libflagship.logincache.guess_region(cache["ab_code"])
-
-    log.info("Initializing API..")
-    appapi = libflagship.httpapi.AnkerHTTPAppApiV1(auth_token=auth_token, region=region, verify=not env.insecure)
-    ppapi = libflagship.httpapi.AnkerHTTPPassportApiV1(auth_token=auth_token, region=region, verify=not env.insecure)
-
-    # request profile and printer list
-    log.info("Requesting profile data..")
-    profile = ppapi.profile()
-
-    # create config object
-    config = cli.model.Config(account=cli.model.Account(
-        auth_token=auth_token,
-        region=region,
-        user_id=profile['user_id'],
-        email=profile["email"],
-    ), printers=[])
-
-    log.info("Requesting printer list..")
-    printers = appapi.query_fdm_list()
-
-    log.info("Requesting pppp keys..")
-    sns = [pr["station_sn"] for pr in printers]
-    dsks = {dsk["station_sn"]: dsk for dsk in appapi.equipment_get_dsk_keys(station_sns=sns)["dsk_keys"]}
-
-    # populate config object with printer list
-    for pr in printers:
-        station_sn = pr["station_sn"]
-        config.printers.append(cli.model.Printer(
-            sn=station_sn,
-            mqtt_key=unhex(pr["secret_key"]),
-            p2p_conn=pr["p2p_conn"],
-            p2p_duid=pr["p2p_did"],
-            p2p_key=dsks[pr["station_sn"]]["dsk_key"],
-        ))
-        log.info(f"Adding printer [{station_sn}]")
+    region = libflagship.logincache.guess_region(cache["ab_code"])
+
+    try:
+        config = cli.config.load_config_from_api(auth_token, region, env.insecure)
+    except libflagship.httpapi.APIError as E:
+        log.critical(f"Config import failed: {E} (auth token might be expired: make sure Ankermake Slicer can connect, then try again")
+    except Exception as E:
+        log.critical(f"Config import failed: {E}")
 
     # save config to json file named `ankerctl/default.json`
     env.config.save("default", config)
 
-    log.info(f"Finished import")
+    log.info("Finished import")
 
 
 @config.command("show")
 @pass_env
 def config_show(env):
     """Show current config"""
 
+    log.info(f"Loading config from {env.config.config_path('default')}")
+    print()
```

Comment:
?

Reply:
For aesthetics :)

When printing the config, the rest of the info is clearly divided into two sections, so this looks nicer, imho.